conans/test/functional/layout/test_clion_layout.py

```diff
@@ -0,0 +1,87 @@
+import os
+
+import pytest
+
+from conans.model.ref import ConanFileReference, PackageReference
+from conans.test.assets.genconanfile import GenConanfile
+from conans.test.utils.tools import TestClient
+
+
+@pytest.fixture
+def conanfile():
+    conanfile = str(GenConanfile().with_import("from conans import tools")
+                    .with_import("import os")
+                    .with_settings("build_type")
+                    .with_import("from conan.tools.layout import clion_layout"))
+
+    conanfile += """
+    def source(self):
+        tools.save("include/myheader.h", "")
+
+    def build(self):
+        tools.save("mylib.lib", "")
+
+    def shape(self):
+        clion_layout(self)
```

Comment:
need a way to override/select/conf by users

Reply:
You can access/change the `self.layout` after that, or what do you mean?